src/security/interfaces.py

```
@@ -50,3 +60,49 @@ def verify_access_token_or_raise(self, token: str) -> None:
         Verify an access token or raise an error if invalid.
         """
         pass
+
+
+class JWTAuthManager(JWTAuthManagerInterface):
+    SECRET_KEY = "mysecretkey"
+    ALGORITHM = "HS256"
+    ACCESS_TOKEN_EXPIRE_MINUTES = 30
+    REFRESH_TOKEN_EXPIRE_DAYS = 7
+
+    def create_refresh_token(
+        self, data: dict, expires_delta: Optional[timedelta] = None
+    ) -> str:
+        to_encode = data.copy()
+        if expires_delta:
+            expire = datetime.utcnow() + expires_delta
+        else:
+            expire = datetime.utcnow() + timedelta(minutes=15)
+        to_encode.update({"expire": expire})
+        encoded_jwt = jwt.encode(to_encode, self.SECRET_KEY, algorithm=self.ALGORITHM)
+        return encoded_jwt
+
+    def decode_access_token(self, token: str) -> dict:
+        try:
+            payload = jwt.decode(token, self.SECRET_KEY, algorithms=[self.ALGORITHM])
+            return payload
+        except JWTError:
+            return None
+
+    def decode_refresh_token(self, token: str) -> dict:
+        try:
+            payload = jwt.decode(token, self.SECRET_KEY, algorithms=[self.ALGORITHM])
+            return payload
+        except JWTError:
```

`decode_refresh_token` has the same problem: it swallows `JWTError` and returns `None` on failure. This can lead to downstream code attempting to use `None` as a dict. Instead, raise the exception (or re-raise a token-specific exception) so token decoding failures are explicit and can be handled by routes (e.g., returning 400).